src/transforms/at-range.js

```diff
@@ -113,14 +113,23 @@ Transforms.deleteAtRange = (transform, range, options = {}) => {
   const endBlock = document.getClosestBlock(document.getNextText(endKey).key)
 
   if (startBlock.key !== endBlock.key) {
-    endBlock.nodes.forEach((child, i) => {
-      const newKey = startBlock.key
-      const newIndex = startBlock.nodes.size + i
+    let blockToRemove = endBlock
+    let blockToKeep = startBlock
+
+    // We keep the style of the blockToKeep. endBlock should keep style if deleting empty line.
+    if (startBlock.isEmpty && !endBlock.isEmpty) {
```

Comment:
So that could be filed as another issue. Let me know if you want to do that. Not sure the level of granularity you are going for.